src/sink/rest.rs

```diff
@@ -391,6 +395,26 @@ impl Sink for Rest {
             Ok(None)
         } else {
             error!("Dropped data due to overload");
+            let error_event = create_error_response(
+                self.error_id_gen.next_id(),
```

Comment:
I am sorry for the back and forth, but upon inspecting the existing code more closely, there is already an `EventIdGenerator` in the codec_task (line 511) and these to would possibly conflict.

I think what should instead be the best solution here is to send `CodecTaskInMsg::ReportFailure` to the `codec_task_channel`. The codec task will then send a `CB::Fail` and an error_response. In this branch here in `on_event` we should not return an `Err` as this would trigger another `CB::Fail` to be sent. Make it an `Ok(None)` in that case.

Again, sorry, I just remembered now how it all works.

Reply:
The rest sink is a beast!

Reply:
Hey sorry for the late reply, uni midterms going on 😩 
Sending `CodecTaskInMsg::ReportFailure` to `codec_task_channel` sounds good to me except for the fact that, the implementation of `codec_task()` always sends a 503 as the status code, which is _technically_ wrong here, since [429 is the status code for too many requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429).
Are we fine with sending 503 as of now, or should we think about modifying the `CodecTaskInMsg::ReportFailure` type to have a `u16` for status codes?

Reply:
Yeah, i think you are right, we should definitely return a 429 to the client.
Adding a status field to the `CodecTaskInMsg::ReportFailure` sounds like the right thing to do :).
Great suggestion.